src/materials/Material.js

```diff
@@ -7,6 +7,20 @@ let _materialId = 0;
 
 class Material extends EventDispatcher {
 
+	static get type() {
+
+		return 'Material';
+
+	}
+
+	get type() {
+
+		return this.constructor.type;
+
+	}
+
+	set type( _value ) { /* */ }
```

Comment:
Not a big fan of this but since we're not using es2020 we can't do:

class Material extends EventDispatcher {
    static type = 'MeshBasicMaterial';
    ...
}

and `MeshBasicMaterial.type = 'MeshBasicMaterial';` would break treeshaking I believe.